src/mlpack/methods/preprocess/preprocess_describe_main.cpp

```diff
@@ -0,0 +1,231 @@
+  double n = input.n_elem;
+  if (population)
+  {
+    // Calculate Population Excess Kurtosis
+    double M2 = SumNthPowerDeviations(input, rowMean, 2);
+    kurtosis = n * (M4 / pow(M2, 2)) - 3;
+  }
+  else
+  {
+    // Calculate Sample Excess Kurtosis
+    double S4 = pow(rowStd, 4);
+    double norm3 = (3 * (n-1) * (n-1)) / ((n-2) * (n-3));
+    double normC = (n * (n+1))/((n-1) * (n-2) * (n-3));
+    double normM = M4 / S4;
+    kurtosis = normC * normM - norm3;
+  }
+  return kurtosis;
+}
+/**
+ * Calculates standard error of standard deviation.
+ *
+ * @param input Vector that captures a dimension of a dataset
+ * @param rowStd Standard Deviation of the given vector.
+ * @return Standard error of the stanrdard devation of the given vector.
+ */
+double StandardError(const arma::rowvec& input, const double rowStd)
+{
+   return rowStd / sqrt(input.n_elem);
+}
+
+int main(int argc, char** argv)
+{
+  // Parse command line options.
+  CLI::ParseCommandLine(argc, argv);
+  const string inputFile = CLI::GetParam<string>("input_file");
+  const size_t dimension = static_cast<size_t>(CLI::GetParam<int>("dimension"));
+  const size_t precision = static_cast<size_t>(CLI::GetParam<int>("precision"));
+  const size_t width = static_cast<size_t>(CLI::GetParam<int>("width"));
+  const bool population = CLI::HasParam("population");
+
+  // Load the data
+  arma::mat data;
+  data::Load(inputFile, data, false, true /*transpose*/);
+
+  // Generate boost format recipe.
+  const string widthPrecision("%-"+
+    to_string(width)+ "." +
+    to_string(precision));
+  const string widthOnly("%-"+
+    to_string(width)+ ".");
+  string stringFormat = "";
+  string numberFormat = "";
+  for (size_t i = 0; i < 11; ++i)
+  {
+    stringFormat += widthOnly + "s";
+    numberFormat += widthPrecision + "f";
+  }
+
+  Timer::Start("statistics");
+  // Headers
+  Log::Info << boost::format(stringFormat)
+      % "dim" % "var" % "mean" % "std" % "median" % "min" % "max"
+      % "range" % "skew" % "kurt" % "SE" << endl;
+
+  // If the user specified dimension, describe statistics of the given
+  // dimension. If it dimension not specified, describe all dimensions.
+  if (CLI::HasParam("dimension"))
+  {
+    // Extract row of the data with the given dimension.
+    arma::rowvec row = data.row(dimension);
+    // These variables are kept for future calculations.
+    double rowMax = arma::max(row);
+    double rowMin = arma::min(row);
+    double rowMean = arma::mean(row);
+    double rowStd = arma::stddev(row, population);
+
+    // Print statistics of the given dimension.
+    Log::Info << boost::format(numberFormat)
+        % dimension
+        % arma::var(row, population)
+        % rowMean
+        % rowStd
+        % arma::median(row)
+        % rowMin
+        % rowMax
+        % (rowMax - rowMin) // range
+        % Skewness(row, rowStd, rowMean, population)
+        % Kurtosis(row, rowStd, rowMean, population)
+        % StandardError(row, rowStd) << endl;
```

Comment:
No problem, I think your solution is perfectly reasonable.  Would you like to open a ticket about the `Log::Info` and `std::setw` incompatibility?  We don't need to solve it now (or soon), but it would maybe be a good task for someone who is looking for ways to contribute, since it's a pretty self-contained and easy-to-replicate problem. :)